src/MOIWrapper.jl

```diff
@@ -0,0 +1,397 @@
+export CbcOptimizer, copy!, optimize!, get
+
+
+using MathOptInterface
+using Cbc.CbcCInterface
+
+const MOI = MathOptInterface
+const MOIU = MathOptInterface.Utilities
+const CbcCI = CbcCInterface
+
+const SUPPORTED_OBJECTIVES = [
+    MOI.ScalarAffineFunction{Float64}
+]
+
+const SUPPORTED_CONSTRAINTS = [
+    (MOI.ScalarAffineFunction{Float64}, MOI.Interval{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.EqualTo{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.LessThan{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.GreaterThan{Float64}),
+    (MOI.SingleVariable, MOI.EqualTo{Float64}),
+    (MOI.SingleVariable, MOI.LessThan{Float64}),
+    (MOI.SingleVariable, MOI.GreaterThan{Float64}),
+    (MOI.SingleVariable, MOI.Interval{Float64}),
+    (MOI.SingleVariable, MOI.ZeroOne),
+    (MOI.SingleVariable, MOI.Integer)
+]
+
+const BOUND_CONSTRAINTS = [
+    (MOI.ScalarAffineFunction{Float64}, MOI.Interval{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.EqualTo{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.LessThan{Float64}),
+    (MOI.ScalarAffineFunction{Float64}, MOI.GreaterThan{Float64}),
+    (MOI.SingleVariable, MOI.EqualTo{Float64}),
+    (MOI.SingleVariable, MOI.LessThan{Float64}),
+    (MOI.SingleVariable, MOI.GreaterThan{Float64}),
+    (MOI.SingleVariable, MOI.Interval{Float64})
+]
+
+const INTEGRALITY_CONSTRAINTS = [
+    (MOI.SingleVariable, MOI.ZeroOne),
+    (MOI.SingleVariable, MOI.Integer)
+]
+
+mutable struct CbcOptimizer <: MOI.AbstractOptimizer
+    inner::CbcModel
+    # env
+    objFunctConstant::Float64
+    CbcOptimizer() = new(CbcModel(), 0.0) # Initializes with an empty model
+end
+
+mutable struct CbcModelFormat
+    nbRows::Int
+    nbCols::Int
+    constraint_matrix::Array{Float64,2}
+    col_lb::Vector{Float64}
+    col_ub::Vector{Float64}
+    obj::Vector{Float64}
+    row_lb::Vector{Float64}
+    row_ub::Vector{Float64}
+    function CbcModelFormat(nbRows::Int, nbCols::Int)
+        col_lb = Vector{Float64}(nbCols)
+        col_ub = Vector{Float64}(nbCols)
+        obj = Vector{Float64}(nbCols)
+        row_lb = Vector{Float64}(nbRows)
+        row_ub = Vector{Float64}(nbRows)
+        constraint_matrix = Array{Float64, 2}(nbRows, nbCols)
+        for rowIdx in 1:nbRows
+            row_lb[rowIdx] = -Inf
+            row_ub[rowIdx] = Inf
+            for colIdx in 1:nbCols
+                constraint_matrix[rowIdx, colIdx] = 0.0
+            end
+        end
+        for colIdx in 1:nbCols
+            col_lb[colIdx] = -Inf
+            col_ub[colIdx] = Inf
+            obj[colIdx] = 0.0
+        end
```

Comment:
Instead of this just use 
```julia
row_lb = fill(-Inf, nbRows)
obj = zeros(Float64, nbCols)
constraint_matrix = sparse(Int[], Int[], Float64[], nbRows, nbCols)  # see note about sparse matrix
```
(Ditto with the other fields.)